byron/ledger/executable-spec/src/Ledger/Update.hs

```diff
@@ -644,6 +646,18 @@ type UPIState =
   , Map UpId Core.Slot
   )
 
+emptyUPIState :: UPIState
+emptyUPIState =
+  ((ProtVer 0 0 0, PParams 0 0 0 0 0.0 0 0 0 0 0 0)
+  , []
+  , Map.empty
+  , Map.empty
+  , Map.empty
+  , Map.empty
+  , Core.PairSet Set.empty
+  , Core.PairSet Set.empty
+  , Map.empty)
+
```

Comment:
Notice that setting the maximum block size to `0` won't allow you to increase it:

![image](https://user-images.githubusercontent.com/175315/57684749-b2b48f80-7636-11e9-853b-d4153ed008ef.png)

![image](https://user-images.githubusercontent.com/175315/57684772-bf38e800-7636-11e9-820e-5f1ea0e16a94.png)



You might want to check the other protocol parameters as well to make sure an update can be proposed (but I think besides the block version you're fine).

FWIW, this is what we have in `cardano-ledger` in `Test.Cardano.Chain.Elaboration.Block`:


```haskell
  gPps        = Update.ProtocolParameters
    { Update.ppScriptVersion    = 0
    , Update.ppSlotDuration     = 0
    , Update.ppMaxBlockSize     = pps ^. maxBkSz
    , Update.ppMaxHeaderSize    = pps ^. maxHdrSz
    , Update.ppMaxTxSize        = 0
    , Update.ppMaxProposalSize  = 0
    , Update.ppMpcThd           = LovelacePortion 0
    , Update.ppHeavyDelThd      = LovelacePortion 0
    , Update.ppUpdateVoteThd    = LovelacePortion 0
    , Update.ppUpdateProposalThd = LovelacePortion 0
    , Update.ppUpdateImplicit   = 0
    , Update.ppSoftforkRule     = Update.SoftforkRule
      (LovelacePortion 0)
      (LovelacePortion 0)
      (LovelacePortion 0)
    , Update.ppTxFeePolicy      = TxFeePolicyTxSizeLinear
      $ TxSizeLinear (mkKnownLovelace @0) (mkKnownLovelace @0)
    , Update.ppUnlockStakeEpoch = 0
    }

```

Reply:
BTW, I'm also thinking why would you need this initial state here ... I would expect that the test code would contain either examples of initial states, or generators for it ...

Reply:
ok, but that just takes the values from `PParams`, right? it is actually those that I am looking for

Reply:
Aye, but I don't think we have an initial PParams, we either generate them or use ad-hoc values if we're doing unit tests ...

Reply:
for now, it really is about getting it to compile, how those numbers will be created is another question, I think. I will go with non-zero maximal sizes for the maximal sizes. The other values have to be chosen by someone who has more insight into this than I have.